WordPress/gutenberg
# Table: Add colors block support

## Description
Replace ad-hoc background color selection for tables with colors block support.

- Allows selection of text and background colors
- With recent feature to skip serialization of block support classes/styles, this can now target the table element as required
- Previous color selections are migrated via a deprecation to the appropriate style attribute
- Includes updated fixture tests 

## How has this been tested?
Manually and via fixtures tests using `npm run test-unit` 

#### Test Instructions
1. Create a new post, add a table block and select a background color for the table
2. Save the post
3. Checkout this PR branch, rebuild, then reload the editor
4. Confirm the table block migrated successfully and the previous color selection is still present
5. Select the table block and modify the background and font colors
6. Save post and confirm selections are rendered correctly on the frontend

## Screenshots 
<img src="https://user-images.githubusercontent.com/60436221/114511998-f299ac80-9c7b-11eb-9da3-7346c74b0c51.gif" width="400"/>

## Types of changes
New feature.

## Checklist:
